### DISCUSSION TITLE
how to get the entity information in the json format

### DISCUSSION THREAD

**Discussion by ashi7upt** [IfcOpenShell/IfcOpenShell#3745] [General]
How can I extract additional information from a parsed file in JSON format using the **entity.get_info()** command without causing new columns to be added to the data frame and without encountering a "Maximum recursion level reached" error when using it within an Azure Function?

**Comment by aothms** [comment]
That's a very specific question with constraints we can't fully comprehend.

> "without causing new columns to be added to the data frame"

which data frame?

> without encountering a "Maximum recursion level reached" error

what triggers the error, how do you call that, on which data?

**Reply by ashi7upt** [reply]
I am posting the code 
        entity_data = []

        for entity in entities:
            entity_dict = {
                "Entity Type": entity.is_a(),
                "name": find_name_map(entity),
                "GlobalId" : entity.GlobalId,
                **entity.get_info(),
                "Parentltid": rel_map.get(entity.GlobalId, None),
                "Properties": find_properties_map(entity),
                "Quantities": find_quantities_map(entity)
            }

            entity_data.append(entity_dict)

        df = pd.DataFrame(entity_data)

**Reply by aothms** [reply]
And do you have a stack trace of the recursion depth? Because it's hard to tell which of these functions overflows the stack.

By definition `**entity.get_info(),` will give you "new columns" (I guess that's what you mean) because of the simple fact that you ingest arbitrary attributes from the IFC schema.

**Reply by ashi7upt** [reply]
That is true. Is it possible for me to get that data in JSON format?
rest of the functions are working fine.

**Reply by aothms** [reply]
`entity.get_info(recursive=True)` will give you nested dictionaries of pure python objects that you feed to `json.dumps()`
